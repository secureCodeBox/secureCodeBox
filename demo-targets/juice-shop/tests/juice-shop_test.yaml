suite: Full Snapshot

tests:
  - it: matches the snapshot
    chart:
      version: 0.0.0
      appVersion: 0.0.0
    set:
      labels: {foo: bar}
      annotations: {foo: bar}
      nameOverride: "foo"
      fullnameOverride: "bar"
      podSecurityContext: {fsGroup: 1234}
      securityContext:
        {
          capabilities: {drop: [ALL]},
          readOnlyRootFilesystem: true,
          runAsNonRoot: true,
          runAsUser: 1234,
        }
      resources: {requests: {cpu: "123m", memory: "123Mi"}}
      ingress:
        enabled: true
        annotations: {foo: bar}
        tls: [{secretName: chart-example-tls, hosts: [chart-example.local]}]
      nodeSelector: {foo: bar}
      tolerations:
        [{key: foo, operator: Exists, value: "value1", effect: NoSchedule}]
      affinity:
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
              - matchExpressions:
                  - {key: foo, operator: In, values: [bar, baz]}
      containerEnv: [{name: foo, value: bar}]
      customConfig:
        {
          server: {port: 3000, basePath: "foo"},
          application: {domain: juice-sh.op, name: "OWASP Juice Shop"},
        }

    asserts:
      - matchSnapshot: {}
