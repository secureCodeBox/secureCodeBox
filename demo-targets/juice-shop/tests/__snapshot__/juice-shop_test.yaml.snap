matches the snapshot:
  1: |
    raw: |2
      1. Get the application URL by running these commands:
        http://chart-example.localmap[path:/]
  2: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        foo: bar
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: juice-shop
        app.kubernetes.io/version: v13.0.3
        foo: bar
        helm.sh/chart: juice-shop-v3.1.0-alpha3
      name: RELEASE-NAME-juice-shop
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: juice-shop
      template:
        metadata:
          annotations:
            foo: bar
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: juice-shop
        spec:
          containers:
            - env: []
              image: docker.io/bkimminich/juice-shop:v13.0.3
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /
                  port: http
              name: juice-shop
              ports:
                - containerPort: 3000
                  name: http
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /
                  port: http
              resources: {}
              securityContext: {}
              volumeMounts: []
          securityContext: {}
          volumes: []
  3: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: juice-shop
        app.kubernetes.io/version: v13.0.3
        foo: bar
        helm.sh/chart: juice-shop-v3.1.0-alpha3
      name: RELEASE-NAME-juice-shop
    spec:
      ingressClassName: nginx
      rules:
        - host: chart-example.local
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME-juice-shop
                    port:
                      number: 3000
                path: /
                pathType: ImplementationSpecific
  4: |
    apiVersion: v1
    kind: Service
    metadata:
      annotations:
        foo: bar
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: juice-shop
        app.kubernetes.io/version: v13.0.3
        foo: bar
        helm.sh/chart: juice-shop-v3.1.0-alpha3
      name: RELEASE-NAME-juice-shop
    spec:
      ports:
        - name: http
          port: 3000
          protocol: TCP
          targetPort: http
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: juice-shop
      type: ClusterIP
  5: |
    apiVersion: v1
    kind: Pod
    metadata:
      annotations:
        helm.sh/hook: test-success
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: juice-shop
        app.kubernetes.io/version: v13.0.3
        foo: bar
        helm.sh/chart: juice-shop-v3.1.0-alpha3
      name: RELEASE-NAME-juice-shop-test-connection
    spec:
      containers:
        - args:
            - RELEASE-NAME-juice-shop:3000
          command:
            - wget
          image: busybox
          name: wget
      restartPolicy: Never
